internal/app/knitgateway/config.go

```diff
@@ -157,33 +204,105 @@ func LoadConfig(ctx context.Context, path string) (*GatewayConfig, error) {
 		}
 	}
 
+	var listenAddress string
+	if extConf.Listen.Port != nil {
+		listenAddress = fmt.Sprintf("%s:%d", extConf.Listen.BindAddress, *extConf.Listen.Port)
+	}
+
 	return &GatewayConfig{
-		ListenAddress: fmt.Sprintf("%s:%d", extConf.Listen.BindAddress, extConf.Listen.Port),
+		ListenAddress: listenAddress,
+		UnixSocket:    extConf.Listen.UnixSocket,
+		TLSConfig:     serverTLS,
 		Services:      services,
 	}, nil
 }
 
 type externalGatewayConfig struct {
-	Listen   externalListenConfig    `yaml:"listen"`
-	Backends []externalBackendConfig `yaml:"backends"`
+	Listen     externalListenConfig     `yaml:"listen"`
+	Backends   []externalBackendConfig  `yaml:"backends"`
+	DefaultTLS *externalClientTLSConfig `yaml:"default_tls"`
```

Comment:
instead of `default_tls` I think a better name could be `upstream_tls` or `client_tls`.

Reply:
I put "default" in the name for clarity. They are just defaults that can be omitted and/or overridden via per-backend configuration. If it's still unclear (for whether it's to configure the incoming or upstream settings), maybe `default_backend_tls`?

Reply:
Done. I ended up going with `backend_tls`, for consistency since the other way to configure is via `backends.tls` stanzas.